rpcserver.go

```diff
@@ -5276,7 +5276,7 @@ func (r *rpcServer) AssetLeaves(ctx context.Context,
 
 // unmarshalLeafKey un-marshals a leaf key from the RPC form.
 func unmarshalLeafKey(key *unirpc.AssetKey) (universe.LeafKey, error) {
-	var leafKey universe.LeafKey
+	var leafKey universe.BaseLeafKey
```

Comment:
Perhaps we can follow the same pattern we used for the `GroupKeyReveal` upgrade from V0 to V1. That would mean introducing a new interface, renaming `universe.LeafKey` to `universe.LeafKeyV0`, and adding a new `universe.LeafKeyV1`, rather than layering changes on top of a shared base. Might help keep the original V0 implementation clearly identifiable.

Reply:
I think the `GroupKeyReveal` case is different, as that changed affected the proof/wire protocol. With this change that you're commenting on, there's no semantic change here: the universe key is untouched. This just positions us to update it later to use the new unique universe key variant. 

For the move, rather than p2p changes, the changes will be based on the RPC interface. We'd add a version to the various queries, then add additional validation based on that version. We'd keep a snapshot of the legacy tree, migrate things over to the new tree, but then only support new _insertions_ into the new tree.